src/relayclient/LightTruffleContract.ts

```diff
@@ -0,0 +1,90 @@
+import Web3 from 'web3'
+import { AbiItem, AbiOutput, toBN } from 'web3-utils'
+import { Web3Provider } from './ContractInteractor'
+
+function retypeItem (abiOutput: AbiOutput, ret: any): any {
+  if (abiOutput.type.includes('int')) {
+    return toBN(ret)
+  } else {
+    return ret
+  }
+}
+
+// restore TF type: uint are returned as string in web3, and as BN in TF.
+function retype (outputs?: AbiOutput[], ret?: any): any {
+  if (outputs?.length === 1) {
+    return retypeItem(outputs[0], ret)
+  } else {
+    // seems like structure return values in truffle are left as strings,
+    return ret
+  }
+}
+
+export class Contract<T> {
+  web3!: Web3
+
+  constructor (readonly contractName: string, readonly abi: AbiItem[]) {
+  }
+
+  createContract (address: string): any {
+    return new this.web3.eth.Contract(this.abi, address)
```

Comment:
let's keep the 'code size' check logic in create contract? I came to rely on it instead of checking manually anywhere

Reply:
we always query contracts for their version. why duplicate (and pull entire bytecode) each time?